backend-rust/src/graphql_api/transaction.rs

```diff
@@ -302,14 +309,26 @@ impl Success<'_> {
                 start = usize::max(start, new_end);
             }
         }
-        let mut connection = connection::Connection::new(start == 0, end == self.events.len());
-        connection.edges = self.events[start..end]
+
+        let edges: Vec<_> = self.events[start..end]
             .iter()
             .enumerate()
             .map(|(i, event)| connection::Edge::new(i.to_string(), event))
             .collect();
+        let total_count = edges.len();
+        let mut connection: Connection<String, &Event, AdditionalFields> =
+            Connection::with_additional_fields(
+                start == 0,
+                end == self.events.len(),
+                AdditionalFields {
+                    total_count,
+                },
+            );
+        connection.edges = edges;
         Ok(connection)
     }
+
+    async fn total_count(&self) -> u64 { self.events.len() as u64 }
```

Comment:
is this needed? The additional fields will be enough right?

Reply:
Yes indeed